rust/agama-server/src/network/model.rs

```diff
@@ -16,19 +19,38 @@ use thiserror::Error;
 use uuid::Uuid;
 use zbus::zvariant::Value;
 
-#[derive(Default, Clone, Debug)]
+#[derive(PartialEq)]
+pub enum NetworkStateItems {
+    AccessPoints,
+    Devices,
+    Connections,
+    GeneralState,
+}
+
+#[derive(Default, Clone, Debug, utoipa::ToSchema)]
 pub struct NetworkState {
+    pub general_state: GeneralState,
+    pub access_points: Vec<AccessPoint>,
     pub devices: Vec<Device>,
     pub connections: Vec<Connection>,
 }
 
 impl NetworkState {
     /// Returns a NetworkState struct with the given devices and connections.
     ///
+    /// * `general_state`: General network configuration
+    /// * `access_points`: Access points to include in the state.
     /// * `devices`: devices to include in the state.
     /// * `connections`: connections to include in the state.
-    pub fn new(devices: Vec<Device>, connections: Vec<Connection>) -> Self {
+    pub fn new(
```

Comment:
We are starting to add quite some parameters. Perhaps we should think about implementing a Builder struct (especially for tests).